l1-contracts/contracts/state-transition/chain-deps/facets/Admin.sol

```diff
@@ -289,11 +289,22 @@ contract AdminFacet is ZkSyncHyperchainBase, IAdmin {
 
     /// @inheritdoc IAdmin
     function forwardedBridgeClaimFailedBurn(
-        uint256 _chainId,
-        bytes32 _assetInfo,
+        uint256 /* _chainId */,
+        bytes32 /* _assetInfo */,
         address _prevMsgSender,
-        bytes calldata _data
-    ) external payable override onlyBridgehub {}
+        bytes calldata /* _data */
+    ) external payable override onlyBridgehub {
+        require(s.settlementLayer != address(0), "Af: not migrated");
+        require(_prevMsgSender == s.admin, "Af: not chainAdmin");
+        IStateTransitionManager stm = IStateTransitionManager(s.stateTransitionManager);
+
+        uint256 currentProtocolVersion = s.protocolVersion;
+        uint256 protocolVersion = stm.protocolVersion();
```

Comment:
You should unpack data and check only the version from the data vs the local one. The STM may upgrade while the chain is in pending state during migration